examples/generative/Density estimation using RealNVP.py

```diff
@@ -0,0 +1,251 @@
+"""
+
+# Generating the examples
+
+url = "http://cs.joensuu.fi/sipu/datasets/s1.txt"
+path = tf.keras.utils.get_file("dataset", url)
+data = np.loadtxt(path)
+scaler = StandardScaler()
+data = scaler.fit_transform(data)
+
+"""
+## Affine coupling layer
+"""
+
+# Creating a custom layer with keras API
+
+
+nweights = 256
+reg = 0.01
+
+
+def Coupling(input_shape):
+    input = keras.layers.Input(shape=input_shape)
+
+    t_layer_1 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(input)
+    t_layer_2 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_1)
+    t_layer_3 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_2)
+    t_layer_4 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_3)
+    t_layer_5 = keras.layers.Dense(
+        2, activation="linear", kernel_regularizer=regularizers.l2(reg)
+    )(t_layer_4)
+
+    s_layer_1 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(input)
+    s_layer_2 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_1)
+    s_layer_3 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_2)
+    s_layer_4 = keras.layers.Dense(
+        nweights, activation="relu", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_3)
+    s_layer_5 = keras.layers.Dense(
+        2, activation="tanh", kernel_regularizer=regularizers.l2(reg)
+    )(s_layer_4)
+    return keras.Model(inputs=input, outputs=[s_layer_5, t_layer_5])
+
+
+Coupling(2).summary()
+"""
+## Real NVP
+"""
+# class containing the whole set of operations
+class Realnvp(keras.Model):
+    def __init__(self, layers_list, masks, distr):
+        super(Realnvp, self).__init__()
+
+        # number of coupling layers
+        self.num_cl = len(masks)
+        # distribution of the latent space
+        self.distr = distr
+        # masks to divide into 1:d and d+1:D
+        self.masks = masks
+        # s and t custom layers of before
+        self.layers_list = layers_list
+
+    # custom function defining the forward operation
+    def forward(self, y):
+        # log determinant of the forward pass
+        log_det_for = tf.zeros(y.shape[0])
+        x = y
+        for i in range(self.num_cl):
+            x_masked = x * self.masks[i]
+            reversed_mask = 1 - self.masks[i]
+            s, t = self.layers_list[i](x_masked)
+            s *= reversed_mask
+            t *= reversed_mask
+            x = x_masked + reversed_mask * (x * tf.exp(s) + t)
+            log_det_for += tf.reduce_sum(s, [1])
+
+        return x, log_det_for
+
+    # custom function defining the forward operation
+    def inverse(self, x):
```

Comment:
You can refactor `call` so that both `forward` and `inverse` are covered by the same `call` method. They only differ very little